File: MOOC_TF_Study/_5/5_1_4.cs
using System;
using System.Collections.Generic;
using System.Text;
using MOOC_TF_Study.Common;
using NumSharp;
using static Tensorflow.Binding;

namespace MOOC_TF_Study._5
{
    class _5_1_4
    {
        public static void Run()
        {
            tf.reset_default_graph();

            var a = tf.Variable(1, name: "a");
            var b = tf.add(a, 1, name: "b");
            var c = tf.multiply(b, 4, name:"c");
            var d = tf.subtract(c, b, name: "d");

            var logdir = "r:/log";

            var writer = tf.summary.FileWriter(logdir, tf.get_default_graph());
            // 没close
        }
    }
}

File: TF_Study_PythonCode/5/5.1.4.py
import tensorflow as tf

# tf 1.5 的 reset_default_graph() 方法，和原教材的版本不一致了
tf.compat.v1.reset_default_graph()

a = tf.Variable( 1, name="a")
b = tf.add(a, 1, name="b")
c = tf.multiply(b, 4, name="c")
d = tf.subtract(c, b, name="d")

logdir = 'r:/log'
writer = tf.compat.v1.summary.FileWriter(logdir, tf.compat.v1.get_default_graph())
writer.close()
